go-controller/pkg/node/ovn_db_mgmt.go

```diff
@@ -0,0 +1,139 @@
+package node
+
+import (
+	"regexp"
+	"strings"
+	"time"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/metrics"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+func (n *OvnNode) ensureOvnRaft(db string) {
+	ticker := time.NewTicker(60 * time.Second)
+	klog.Infof("Starting ensure routine for Raft db: %s", db)
+	kclient := n.Kube.(*kube.Kube)
+	for {
+		select {
+		case <-ticker.C:
+			if dbOnNode, _ := metrics.CheckPodRunsOnGivenNode(kclient.KClient.(*kubernetes.Clientset),
+				"name=ovnkube-db", n.name, false); dbOnNode {
+				klog.Infof("Node %s has a OVN DB pod, ensuring OVN Raft membership", n.name)
+				n.ensureLocalRaftServerID(db)
+				n.ensureClusterRaftMembership(db)
+			}
```

Comment:
There's a couple options here. I could just make it only check 1 time (at startup) if this node has a db co-located. It would handle the case of adding a new node with a db, but would not handle if a db pod was removed or added to an existing node. Doesn't seem great.

@girishmg what if we make it into its own go binary and run it as part of the ovnkube-db container via ovnkube.sh? Is there a reason you think it needs to be a separate container? I wonder if then we could move a lot of the other bash into golang here as well...

Reply:
Ideally, we would need a OVSDB operator to remediate things correctly. You cannot fix RAFT by running commands on a node. These commands will work only if the RAFT is healthy (has majority number of nodes healthy).

Yes, keeping all these changes in ovnkube.sh itself works. We don't need separate container. I would rather just keep all this in the script itself instead of a go binary. YMMW

Reply:
I think maybe we need to have a meet to discuss this stuff. I'm not sure I explained what this was fixing well enough, and I'd like to come to a consensus on where this stuff should run. I'll ping you on slack.